File: C++/pertemuan9/latihanprogramkasir.cpp
#include <iostream>

using namespace std;

int main() {
    string namaBarang;
    int harga, jumlah;
    double diskon, total, totalbayar;

    cout << "Masukkan nama barang: ";
    cin >> namaBarang; 
 
    cout << "Masukkan harga barang: ";
    cin >> harga;


    cout << "Masukkan jumlah barang: ";
    cin >> jumlah;


    total = harga * jumlah;

 
    if (total >= 39300) {
        diskon = total * 0.03; // 3%
    } else {
        diskon = 0;
    }


    totalbayar = total - diskon;

 
    cout << "\n======== Struk Pembelian ========\n";
    cout << "Nama Barang   : " << namaBarang << endl;
    cout << "Harga Barang  : " << harga << endl;
    cout << "Jumlah Barang : " << jumlah << endl;
    cout << "Total         : " << total << endl;
    cout << "Diskon        : " << diskon << endl;
    cout << "Total Bayar   : " << totalbayar << endl;
    cout << "=================================\n";

    return 0;
}

File: JAVA/pertemuan9/latihanprogramkasir.java
package pertemuan9;

import java.util.Scanner;

public class latihanprogramkasir {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.println("Masukan Nama Barang:");
        String nama = sc.nextLine();
        System.out.println("Masukan Harga Satuan:");
        double harga = sc.nextDouble();
        System.out.println("Masukan Jumlah Beli:");
        int jumlah = sc.nextInt();
    
        double total = harga * jumlah;
        double diskon;

        if (total > 39300) {
            diskon = total * 0.03;
        } else {
            diskon = 0;
        }

        double totalbayar = total - diskon;

        System.out.println("\n=========STRUK PEMBELIAN=========");
        System.out.println("Nama Barang   : " + nama);
        System.out.println("Harga Satuan  : " + harga);
        System.out.println("Jumlah Beli   : " + jumlah);
        System.out.println("Total Harga   : " + total);
        System.out.println("Diskon        : " + diskon);
        System.out.println("Total Bayar   : " + totalbayar);
        System.out.println("=================================");
    }
}
